htdocs/user/card.php

```diff
@@ -450,11 +450,15 @@
 				// Do we update also ->entity ?
 				if (!empty($conf->multicompany->enabled)) {	// If multicompany is not enabled, we never update the entity of a user.
 					if (!empty($conf->global->MULTICOMPANY_TRANSVERSE_MODE)) {
-						$object->entity = 1; // all users are in master entity
+						if (GETPOST('superadmin', 'int')) {
+							$object->entity = 0;
```

Comment:
Relying on a property provided as a parameter to decide if an object must be into an admin entity or not means anybody, even not superadmin can set the superadmin flag to any user.

Reply:
@eldy 
I check this in the update function of the class user